File: SundeeFundee/Domain/WODTemplateType.swift
import Foundation

enum WODTemplateType: String, Sendable {
    case strength
    case amrap
    case emom
    case forTime
    case circuit

    static func from(_ rawValue: String) -> WODTemplateType {
        WODTemplateType(rawValue: rawValue) ?? .strength
    }

    var displayName: String {
        switch self {
        case .strength: "Strength"
        case .amrap: "AMRAP"
        case .emom: "EMOM"
        case .forTime: "For Time"
        case .circuit: "Circuit"
        }
    }

    var requiresTimer: Bool {
        switch self {
        case .amrap, .emom, .forTime: true
        case .strength, .circuit: false
        }
    }
}

File: src/domain/shared/wod-template-type.ts
/**
 * WODTemplateType
 * Ported from SundeeFundee/Domain/WODTemplateType.swift
 *
 * Workout-of-the-day template categories.
 * Pure type + pure helper functions.
 */

// ---------------------------------------------------------------------------
// WODTemplateType — string union matching Swift enum
// ---------------------------------------------------------------------------

export type WODTemplateType = 'strength' | 'amrap' | 'emom' | 'forTime' | 'circuit';

// ---------------------------------------------------------------------------
// Pure functions
// ---------------------------------------------------------------------------

/**
 * Parse a raw string value to a WODTemplateType.
 * Defaults to 'strength' for unknown values.
 * Matches Swift WODTemplateType.from(_:).
 */
export function wodTemplateTypeFrom(rawValue: string): WODTemplateType {
  switch (rawValue) {
    case 'strength': return 'strength';
    case 'amrap': return 'amrap';
    case 'emom': return 'emom';
    case 'forTime': return 'forTime';
    case 'circuit': return 'circuit';
    default: return 'strength';
  }
}

/**
 * Human-readable display name.
 * Matches Swift WODTemplateType.displayName computed property.
 */
export function wodTemplateTypeDisplayName(type: WODTemplateType): string {
  switch (type) {
    case 'strength': return 'Strength';
    case 'amrap': return 'AMRAP';
    case 'emom': return 'EMOM';
    case 'forTime': return 'For Time';
    case 'circuit': return 'Circuit';
  }
}

/**
 * Whether this template type requires an active timer during the workout.
 * Matches Swift WODTemplateType.requiresTimer computed property.
 */
export function wodTemplateTypeRequiresTimer(type: WODTemplateType): boolean {
  switch (type) {
    case 'amrap':
    case 'emom':
    case 'forTime':
      return true;
    case 'strength':
    case 'circuit':
      return false;
  }
}
